sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -333,6 +333,16 @@ def _parse_odata_body(error_format, response):
             pass
         return None
 
+    def __str__(self):
+        retval = super(HttpResponseError, self).__str__()
+        try:
+            body = self.response.text()
+            if body and not self.error:
```

Comment:
So this is the check for the odata error? `self.error` will be empty if this we weren't able to parse the odata?

Reply:
Yes.